### DISCUSSION TITLE
[BUG] findAllListingsActive sort_on: updated desc – not correctly sorted

### DISCUSSION THREAD

**Discussion by sampok** [etsy/open-api#905] [Q&A]
The [`findAllListingsActive`](https://developer.etsy.com/documentation/reference/#operation/findAllListingsActive) endpoint allows for sorting on `"updated"` in `"desc"` order. However the `updated_timestamp` is not monotonically descreasing. This makes it hard to poll and follow updated listings as it's unclear how deep in pagination some 'unseen' listings might be.

## Steps to reproduce

Running this with Node.js...

```js
// Request active listings sorted by updated desc
let { results } = await fetch(
  "https://openapi.etsy.com/v3/application/listings/active?" +
    new URLSearchParams({
      sort_on: "updated",
      sort_order: "desc",
      taxonomy_id: 374,
    }),
  { headers: { "x-api-key": "[REDACTED]" } }
).then((response) => response.json());

// Calculate the diff between each listing's updated timestamp
// assuming it to be monotonically decreasing
let listings = results.map((l, index) => {
  const prevListing = results[index - 1];
  const diff = prevListing
    ? l.updated_timestamp - prevListing.updated_timestamp
    : "n/a";

  return {
    listing_id: l.listing_id,
    timestamp: l.updated_timestamp,
    diff: String(diff).padStart(4),
    note: diff > 0 ? "Unexpected positive diff!" : "",
  };
});

console.log(Object.keys(listings[0]).join("\t"));
console.log(listings.map((l) => Object.values(l).join("\t")).join("\n"));
```

... results in:

```
listing_id      timestamp       diff    note
1156947456      1678630511       n/a
1437017621      1678630502        -9
1315940412      1678630492       -10
1437017585      1678630492         0
1428383803      1678630489        -3
1180396931      1678630488        -1
1082337708      1678630487        -1
1151645928      1678630552        65    Unexpected positive diff!
1371620975      1678630477       -75
1421384619      1678630476        -1
1413994878      1678630476         0
1064886882      1678630466       -10
1040135069      1678630462        -4
1437017355      1678630464         2    Unexpected positive diff!
1395024925      1678630462        -2
1330207420      1678630460        -2
1343586777      1678630460         0
1343832297      1678630459        -1
1407343916      1678630455        -4
735326860       1678630453        -2
749200497       1678630453         0
880753400       1678630452        -1
1437008489      1678630453         1    Unexpected positive diff!
880736544       1678630452        -1
876233524       1678630450        -2
```

## Expected Behavior
The returned listings should be in monotonically descending updated order as expected from the documentation.

**Comment by openapi-support** [comment]
Thank you for reporting this issue. We can confirm that sorting by updated timestamp (`sort_on=updated`) is currently not supported in our API, despite being documented as available.

* **Current Status**: We're investigating the implementation of this feature. Our technical analysis has identified that this requires infrastructure updates to support this sorting capability.
* **Technical Context**: While our API accepts the `sort_on=updated` parameter, the feature is not fully implemented. This explains why requests fall back to creation date sorting instead.
* **Next Steps**: We're working with our engineering teams to determine the implementation timeline and technical requirements.

We'll update this issue with when the problem is resolved.
